feat: add js bindings to /v2/monitors

ThetaBird opened the pull request:
Adding js bindings to /v2/monitors as discussed in 
- #263 

The changes in this PR closely follow the conventions established throughout the rest of the codebase.

Checklist: 
 - [x] Add JS bindings to /v2/monitors
 - [x] Add monitor unit tests `(test.ts)`
 - [x] Add monitor integration tests `(spec.ts)`
 - [x] Add monitor js usage examples in `/examples/js/src`
 - [x] Manually validate js bindings in Axiom

gabrielelpidio commented on packages/js/src/monitors.ts:
```diff
@@ -0,0 +1,75 @@
+    disabledUntil?: string;
+  }
+
+  export interface CreateRequest {
```
Nit: For this type you can do something like:
```ts
export interface CreateRequest extends Omit<Monitor, 'id' | 'createdAt' | 'createdBy'> {}
```
Should help avoid mistakes/typos in the future if this gets modified

ThetaBird replied:
Revised

gabrielelpidio commented on packages/js/src/monitors.ts:
```diff
@@ -0,0 +1,75 @@
+    disabledUntil?: string;
+  }
+
+  export interface UpdateRequest {
```
Same here, you should be able to extend `CreateRequest` with
```ts
export interface UpdateRequest extends Partial<CreateRequest> {}
```

ThetaBird replied:
The PUT request for a monitor requires all fields to be defined - otherwise fetch fails with errors like these:

`Error: incorrect operator must be one of [Above Below AboveOrEqual BelowOrEqual]`
`Error: line: 1, col: 1: unexpected end of input (expected (<ident> | ("[" <string>+ "]")) "*"?)`

But yeah, my types for that were incorrectly set to optional. Will revise to match `CreateRequest`

gabrielelpidio commented on packages/js/src/monitors.ts:
```diff
@@ -0,0 +1,75 @@
+    notifyByGroup: boolean;
+    resolvable?: boolean;
+    notifierIDs: string[];
+    intervalMinutes: number;
+    rangeMinutes: number;
```
I think we should leave these as `interval` and `range` and match the GO spec
https://github.com/axiomhq/axiom-go/blob/0720dba1b5b107c0a1709b65cf6a2d9df3589a78/axiom/monitors.go#L138-L140

ThetaBird replied:
It seems like @axiomhq/js is currently trying to match the API spec:

Case in point: `datasets`
- [go](https://github.com/axiomhq/axiom-go/blob/0720dba1b5b107c0a1709b65cf6a2d9df3589a78/axiom/datasets.go#L79-L82) uses `CreatedBy` and `CreatedAt` to map `who` and `created` from API
- [js](https://github.com/axiomhq/axiom-js/blob/980a5c9d333c65bf40beaee5d54487c15d31b100/packages/js/src/datasets.ts#L10-L11) uses `who` and `created` directly

I'm not strongly opinionated on which way to go - just want to make sure that we're either consistent or in agreement of a vision where we are consistent in the future. Unfortunately, mapping existing js structures to match the go spec would cause breaking changes in package upgrades, and vice versa :(

ThetaBird commented:
Hey @gabrielelpidio, just following up on this PR - I addressed some of your comments and responded to others. Is there any more work that needs to be done to get this in? Ty!

gabrielelpidio commented:
@ThetaBird Thanks a lot for this fantastic contribution! 

I'm going to merge it and release it afterward!
